### PR TITLE
Adding option for displaying non-cropped hero image in post

### PR DESCRIPTION
**Pull request by slightlyoffbeat**
Problem: Marketing is creating hero images (featured images) for blog posts that should not be cropped or truncated in the blog post. [See this example](https://blog.mozilla.org/blog/2017/09/26/firefox-quantum-beta-developer-edition/). The featured image is important because it gets used for facebook, twitter and press. 

Solution: Provide an option (custom field) for how to display the hero image. Option 1: Passive. This is the current behavior and would be the default. Option 2: Featured. This would display a non-truncated version of the featured image instead.

This pull request creates that custom field using ACF.

- Requires [ACF](https://www.advancedcustomfields.com/)
- Requires correct field groups. Import file can be obtained from Dan Brown

### CODE REVIEW

**Comment by slightlyoffbeat** [comment]
JSON file to import to get correct ACF fields for this PR: https://pastebin.com/0WQA2pbw

**Comment by craigcook** [comment]
Is there a way to implement this feature without requiring ACF? I really really really don't like the idea of requiring such a complex plugin for this one fairly small enhancement.

Other alternatives that come to mind:

1. Bake the option directly into the theme without a third-party plugin. We could still use a custom field but not rely on ACF to manage it.
2. Convince marketing to stop making featured images that can't be cropped. They're supposed to be optional images for added flavor, not essential content.
3. Rework the design so featured images are never cropped. We opted to crop them on the single post view because we wanted to preserve the 16:9 ratio of the original, but also not push the content so far down the page when the image is really big. We could think about minimizing the main header on the single page, or just making the featured image narrower so even at full 16:9 it's not so tall.

#1 will take a little more effort but avoids the plugin dependency. #2 is probably a lost cause at this point. I favor #3 as a design solution vs a more complex technical solution. 

When we first created this theme the featured images were never intended to be real content, only decoration. Some stock photo or even just a pattern, something to set the mood for the post but the post would stand on its own with no image at all. But once it was in the hands of real users they instantly started treating it as a giant headline and putting vital content in a place where it shouldn't be. So maybe we just need to design for that real-world use case rather than try to retrain the users or introduce new options that make it even harder to use. If they really want text banners on posts then let's try to design posts that can accommodate text banners.

Another option I'm hesitant to even mention is sort of a combination of #2 and #3... if we change the design so featured images aren't so prominent, that would discourage authors from using it as a banner. Instead of trying to retrain them to make better banners, we could just take their banners away. We can still have featured images but in a way that makes them impossible to misuse. I'm not sure what that might look like but I'd be keen to explore it.

So tl;dr - I'm gonna say no to the ACF requirement. I'm open to the idea of building the option in directly, but would like to at least look into reworking the page layout before we go that route. Happy to get on vidyo to throw some ideas around.

**Comment by slightlyoffbeat** [comment]
I understand being hesitant to bring in a plugin for a solution such as this. I'll state my case, but I'll also speak to some of the alternatives. Hopefully, we'll find a solution somewhere.

I generally hate Wordpress plugins as they are usually unnecessary, broken, or poorly implemented. My general rule is "don't use Wordpress plugins" but ACF is always my one exception. ACF is one of the only WP plugins that I believe actually helps the WordPress development process. It is certainly possible to add a custom field without ACF, but maintaining it, or adding to it later can become problematic. ACF, on the other hand, is a trusted plugin that is well documented. I'd argue that ACF is a better, cleaner, and more scalable solution than baking it into the theme. Still, if that is one of the ways of getting this done, I can look into it.

I'd agree with #2 except that the featured image is what users see on facebook and twitter. Take a look at [this](https://twitter.com/search?q=quantum%20beta%20firefox&src=typd) search result on Twitter. The chart and images really helped people understand the feature benefits, and helped drive sharing, retweets, link clicks, etc. It is also what press and blogs grab and use, even if you tell them not to. The image that is used in the metadata for facebook sharing and twitter cards is an important marketing tool, so I want to make sure that marketing can utilize and harness this. Asking marketing to create an image that is used by facebook, twitter, press and also the blog (but is cropped/truncated very differently at every breakpoint) is a large ask.

#3 is easy to do, and would just require a few small changes to the CSS. We'd have to check and see if there are any issues applying this retroactively to all existing blog posts however. I think it'd be better to provide a choice. Either have a passive image where cropping is no big deal or have a more featured image where time and energy is spent on that image and it shouldn't be cropped.

I'll set up 15 min on vidyo. I'm sure we can agree on one of the proposed solutions and I can resubmit if needed.

**Comment by craigcook** [comment]
After further discussion we've decided to build an option into the theme (to sidestep the ACF dependency) letting authors choose to display the image on single posts. The default will be no image on single posts so it's opt-in rather than opt-out. The featured image still gets used for social and shown in the grid view, it just won't be shown on single posts unless the author really wants to show it.

If they do opt to display a hero image, we'll show it at the full 16:9 ratio. That means we still have the issue of a giant image filling the window and pushing content further below the fold. To combat that we can just make the whole thing smaller (but still pretty big). I suggest adding a new thumbnail size called `post-extra-large` at 1000 x 550 for use as the hero image (we can run Regenerate Thumbnails once it lands). Depending on how it looks in layout we could further size that down to `max-width: 940px;` to match the content column.
